parsers/NTESMO.py

```diff
@@ -0,0 +1,214 @@
+        raise Exception(
+            f"Cannot find file on the index page for date {target_datetime}"
+        )
+    return extraction_func(data_file)
+
+
+def parse_consumption(
+    raw_consumption: pd.DataFrame,
+    target_datetime: datetime,
+    logger: Logger,
+    price: bool = False,
+) -> List[dict]:
+    data_points = []
+    for _, consumption in raw_consumption.iterrows():
+        # Market day starts at 4:30 and reports up until 4:00 the next day.
+        # Therefore timestamps between 0:00 and 4:30 excluded need to have an extra day.
+        raw_timestamp = consumption[0]
+        timestamp = datetime.combine(date=target_datetime.date(), time=raw_timestamp)
+        if raw_timestamp < time(hour=4, minute=30):
+            timestamp = timestamp + timedelta(days=1)
+
+        data_point = {
+            "zoneKey": "AU-NT",
+            "datetime": timestamp,
+            "source": "ntesmo.com.au",
+        }
+        if price:
+            data_point["price"] = consumption["Market Price"]
+            data_point["currency"] = "AUD"
+        else:
+            data_point["consumption"] = consumption["Demand"]
+        data_points.append(data_point)
+    return data_points
+
+
+def parse_production_mix(
+    raw_production_mix: pd.DataFrame, logger: Logger
+) -> List[dict]:
+    production_mix = []
+    generation_units = set(raw_production_mix.columns)
+    generation_units.remove("Period Start")
+    generation_units.remove("Period End")
+    if not generation_units == PLANT_MAPPING.keys():
+        raise Exception(
+            f"New generator {generation_units - PLANT_MAPPING.keys()} detected in AU-NT, please update the mapping of generators."
+        )
+    for _, production in raw_production_mix.iterrows():
+        data_point = {
+            "zoneKey": "AU-NT",
+            "datetime": production["Period Start"],
+            "source": "ntesmo.com.au",
+            "production": dict(),
+            "storage": dict(),
+        }
+        for generator_key, generator in PLANT_MAPPING.items():
+            if generator_key not in production:
+                raise Exception(
+                    f"Missing generator {generator_key} detected in AU-NT, please update the mapping of generators."
+                )
+            # Some decomissioned plants have negative production values.
+            if production[generator_key] >= 0:
+                if generator["fuel_type"] in data_point["production"]:
+                    data_point["production"][generator["fuel_type"]] += production[
+                        generator_key
+                    ]
+                else:
+                    data_point["production"][generator["fuel_type"]] = production[
+                        generator_key
+                    ]
+        production_mix.append(data_point)
+    return production_mix
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_consumption(
+    zone_key: str = "AU-NT",
+    session: Optional[Session] = None,
```

Comment:
```suggestion
    session: Session = Session(),
```
Just like `fetch_production_mix`.